### ISSUE TITLE
Estimated time remaining in simulation

### ISSUE THREAD

**Issue by asoplata** [DynaSim/DynaSim#195]
Didn't DNSIM have something that output when a certain percentage of the simulation was done, and/or functionality to tell you how long the simulation was expected to take? Is this difficult to reactivate/reimplement?

Several of my DynaSim benchmarks hit the 12 hour limit, and i can resubmit them to last for longer, but the point is that I have no idea how much longer they're expected to take.

**Comment by kupiqu** [comment]
Yes, it showed the progress of a simulation in 1/5 parts or so, but it only worked for non-coder simulations (the coder was not happy with fprintf/disp calls), but I think I can workaround this (I did it with a save call). Let me look at it...

**Comment by kupiqu** [comment]
Ok, finally I got some spare time to do this.

I'll describe how you have to apply it to dynasim (sorry, but I am not familiar with the dynasim source code yet):

1. In new version of dnsimulator (the function that creates the odefun file) you'll need to specify that the function enableLog (it will be showing the output you want) is extrinsic (i.e., it will be run in matlab):

`fprintf(fid,'coder.extrinsic(''enableLog'');\n');`

2. Also in dnsimulator, you'll need to call that function at each time step (it can appear last, before the 'end' of the for loop):

`fprintf(fid,'  tstart = enableLog(k,logTimes,T,tstart);\n');`

3. Finally, you need to create the function itself:

```
function tstart = enableLog(k,logTimes,T,tstart)
% if called, odefun will inform about partial elapsed time at regular periods of the simulation

if tstart == 0
  tstart = tic;
elseif any(k == logTimes)
  elapsedTime = toc(tstart);
  elapsedTimeMinutes = floor(elapsedTime/60);
  elapsedTimeSeconds = rem(elapsedTime,60);
  if elapsedTimeMinutes
    logMS = 'Processed %g of %g ms (elapsed time: %g m %.3f s)\n';
    fprintf(logMS,T(k),T(end),elapsedTimeMinutes,elapsedTimeSeconds);
  else
    logS = 'Processed %g of %g ms (elapsed time: %.3f s)\n';
    fprintf(logS,T(k),T(end),elapsedTimeSeconds);
  end
end
```

That's all... Hope it helps! Let me know if you encounter any problem...

PS I tested it with and without the coder in dnsim and works, so it should in dynasim.

**Comment by kupiqu** [comment]
Jason or Austin, would you be interested in add this functionality? I could help if sth above is not clear...
